File: src/Recursion/Factorial.cpp
#include <iostream>

int factorial(int n)
{
    if (n == 0)
    {
        return 1;
    }
    else 
    {
        return n * factorial(n - 1);
    }
}

int main()
{
    std::cout << "Recursion Factorial" << std::endl;
    int n = 5;
    int result = factorial(n);
    std::cout << "n = " << n << std::endl;
    std::cout << "Factorial of " << n << " is: " << result << std::endl;
    return 0;
}

File: src/Recursion/Factorial.py
class Solution:
    def Factorial(self, n):
        if n == 0:
            return 1
        else:
            return n * self.Factorial(n - 1)


def main():
    sol = Solution()
    print("result: ", sol.Factorial(5))


if __name__ == "__main__":
    main()
